content/doc/book/using/remote-access-api.adoc

```diff
@@ -0,0 +1,260 @@
+visiting http://ci.jruby.org/api/ will show just the top-level API
+features available – primarily a listing of the configured jobs for this
+Jenkins instance. +
+Or if you want to access information about a particular build, e.g.
+http://ci.jruby.org/job/jruby-base/lastSuccessfulBuild/, then go to
+http://ci.jruby.org/job/jruby-base/lastSuccessfulBuild/api/ and you'll
+see the list of functionalities for that build.
+
+The work on this front is ongoing, so if you find missing features,
+please file an issue.
+
+[[RemoteaccessAPI-Whatcanyoudowithit]]
+== What can you do with it?
+
+Remote API can be used to do things like these:
+
+. retrieve information from Jenkins for programmatic consumption.
+. trigger a new build
+. create/copy jobs
+
+[[RemoteaccessAPI-Submittingjobs]]
+== Submitting jobs
+
+*Jobs without parameters*
+
+You merely need to perform an HTTP POST on
+`+JENKINS_URL/job/JOBNAME/build+`.
+
+*Jobs with parameters*
+
+Also see
+https://wiki.jenkins.io/display/JENKINS/Parameterized+Build[Parameterized
+Build].
+
+Simple example - sending "String Parameters":
+
+[source,sh]
+----
+curl -X POST JENKINS_URL/job/JOB_NAME/build \
+  --user USER:TOKEN \
+  --data-urlencode json='{"parameter": [{"name":"id", "value":"123"}, {"name":"verbosity", "value":"high"}]}'
+----
+
+Another example - sending a "File Parameter":
+
+[source,sh]
+----
+curl -X POST JENKINS_URL/job/JOB_NAME/build \
+  --user USER:PASSWORD \
+  --form file0=@PATH_TO_FILE \
+  --form json='{"parameter": [{"name":"FILE_LOCATION_AS_SET_IN_JENKINS", "file":"file0"}]}'
+
+E.g.curl -X POST http://JENKINS_URL/job/JOB_NAME/build  --form file0=@/home/user/Desktop/sample.xml --form json='{"parameter": [{"name":"harness/Task.xml", "file":"file0"}]}'
+Please note, in this example, the symbol '@' is important to mention. Also, the path to the file is absolute path.
+In order to make this command work, you need to configure your Jenkins job to take a file parameter and 'name' in this command corresponds to 'file location' field in the Jenkins job configuration.
+
+In above example, 'harness' is just a name of folder that may not exist in your workspace, you can write "name":"Task.xml" and it will place the Task.xml at root of your workspace.
+Remember that name in this command should match with File location in file parameter in job configuration.
+----
+
+image:../images/using/remote-access-api/file-parameter.png[image,title="File Parameter"]
+
+[[RemoteaccessAPI-RemoteAPIandsecurity]]
+== Remote API and security
+
+When your Jenkins is secured, you can use HTTP BASIC authentication to
+authenticate remote API requests. See
+https://wiki.jenkins.io/display/JENKINS/Authenticating+scripted+clients[Authenticating
+scripted clients] for more details.
+
+[[RemoteaccessAPI-CSRFProtection]]
+== CSRF Protection
+
+*Note*: From Jenkins 2.96 onward, you can use an API token and avoid
+using a crumb / CSRF token.
+
+If your Jenkins uses the "Prevent Cross Site Request Forgery exploits"
```

Comment:
you can no longer turn off csrf prevention now so this sentence isn't needed.

I'm not sure if crumb mentioning is needed at all,

API tokens should be used instead